win32ss/drivers/videoprt/registry.c

```diff
@@ -586,6 +586,10 @@ IntCreateNewRegistryPath(
         return Status;
     }
 
+    /* Copy the saved resolution to the source key */
```

Comment:
Could you clarify whether goal is only to preserve these `NewKey` values, or also to actually update `SettingsKey` too?

Reply:
@SergeGautherie, Just below the lines that I added we have the following:
`Status = IntCopyRegistryKey(SettingsKey, NewKey);`
In both functions here and above the first value is the "Source" and the second is the "Destination".
So in this statement the "NewKey" values are about to be overwritten with wrong resolution data ("SettingsKey").
My solution copies the data that had the resolution that we want to keep ("NewKey") to the source key in this statement.
So, when the source key here is copied to the destination ("NewKey") it will carry along the wanted resolution values.
I only want to carry the two resolution keys forward in this manner because other values may be important to the function that @tkreuzer was trying to fix in his previous commit.

Reply:
Line 571: "Failed to open settings key. Status 0x%lx\n" -> "Failed to open new settings key. Status 0x%lx\n" ? 571 vs 584, old copy-pasta?

Reply:
Ok, so you are preserving the data from DefaultSettings.XResolution and DefaultSettings.YResolution values stored in NewKey by copying them into SettingsKey before overwriting NewKey with SettingsKey. Now I gotcha.

Reply:
> I only want to carry the two resolution keys forward

But your code is also actually updating `SettingsKey`, which seems unintended/unwanted, at least unexplained.
In other words, why do both keys need to be updated?

Reply:
> 571 vs 584, old copy-pasta?

True, it is not fully explicit, but correct nonetheless.
Additionally, reusing same string saves some resources.

Reply:
@SergeGautherie, Yes, you are correct in that I am updating the `SettingsKey` and the reason for this is that it is often wrong and contains the default resolution and we want it to be correct. Also, I have now figured out that your comment about needing changes for the ZwOpenKey functions was in regard to the DesiredAccess. I have made changes that now should account for both reading and writing of both values. In the added code, the Source and Destination keys are reversed from what was used previously in the statement below. Therefore, now we are both reading and writing both keys. 
@vicmarcal, I took care of the copy-pasta while here. Please see if you agree with what I am suggesting.
Thanks to both of you.

Reply:
> it is often wrong

You sound as this is a hack and a proper fix for that key might be needed elsewhere/before.
Could you state what these 2 keys actually are, what they are used for, what happens if only one of them is updated (with proper values) here? Thanks.
In other words, are these key/value changes what Windows does?

> both reading and writing of both values

Right!

Reply:
As said, this is a hack. We should never copy from 'new' to 'old' key.
See https://github.com/reactos/reactos/commit/ebd01880a353bbb3d1f110d52086e008900627da for probably a better fix.

Reply:
@hpoussin, I guess that you may not have noticed, but your "better fix" is just a revert of the last commit by @tkreuzer. So this re-introduces the problem with https://jira.reactos.org/browse/CORE-17688. Fixing this problem was the reason that @tkreuzer made this commit and it seemed to work based on feedback from the users. If you are set against copying from 'new' to 'old' key, then I have a new patch for your review in https://jira.reactos.org/browse/CORE-17719 that is called 'video_save_03.patch'. Thanks very much for your investigations and suggestions. I believe that together we will eventually fix this problem. Doug Lyons

Reply:
>  "sometimes it is wrong for our purposes"

That still does not convince me your proposal is the right fix.

--

> ebd0188

It looks like that is basically reverting ecf3416.
Was the latter totally wrong? Or is an in-between solution (See @tkreuzer's commit description) needed?

Reply:
@Doug-Lyons as I just discussed in PM with @hpoussin, it is OK to revert that Timo's commit, since my tests prove that my recent c05a45e17eda057d8b92ba76feecf3532d8324c1 commit, which was a fix for https://jira.reactos.org/browse/CORE-17786 from the start, also fixes https://jira.reactos.org/browse/CORE-17688 properly. :slightly_smiling_face: So @tkreuzer's change is no longer needed and we now can safely revert it, to fix https://jira.reactos.org/browse/CORE-17719 too. :wink: And if your solution is hacky, it now seems no longer needed.
![mm-pm](https://user-images.githubusercontent.com/26385117/145234264-31d2e796-d933-4c29-b0a0-5f900fb512e3.png)